bootstrapping.py

```diff
@@ -3,391 +3,382 @@
+    parser.add_argument(
+        '-v', '--vcf_file', required=True,
+        type=str, action='store',
+        help='Path to the input vcf file.',
+    )
+    parser.add_argument(
+        '-m', '--meta_data', required=True,
+        type=str, action='store',
+        help='Path to the tab delimited text file where the first column contains'\
+        +' the sample name as it appears in the header of the input vcf file and the'\
+        +' second column contains the population the sample belongs to.',
+    )
+    parser.add_argument(
+        '-f', '--frequency',
+        action=argparse.BooleanOptionalAction,
+        help='Enable calculating site patterns from derived allele frequencies,'\
+        +' otherwise site patterns will be calcualted by randomly sampling one chromosome.',
+    )
+    parser.add_argument(
+        '-p1', '--p1_population', required=True,
+        type=str, action='store',
+        help='Population in the meta data file to use as P1 for comparisons'\
+        +' (ie a potential recipient population).',
+    )
+    parser.add_argument(
+        '-p2', '--p2_population', required=True,
+        type=str, action='store',
+        help='Population in the meta data file to use as P2 for comparisons'\
+        +' (ie a potential recipient population).',
+    )
+    parser.add_argument(
+        '-p3', '--p3_population', required=True,
+        type=str, action='store',
+        help='Population in the meta data file to use as P3 for comparisons'\
+        +' (ie the source population).',
+    )
+    parser.add_argument(
+        '-p4', '--p4_population', required=True,
+        type=str, action='store',
+        help='Population in the meta data file to use as P3 for comparisons'\
+        +' (ie the outrgroup population used for polarization).',
+    )
+    parser.add_argument(
+        '-cl', '--contig_length', required=True,
+        type=int, action='store',
+        help='Total length in base pairs of conting.'
+    )
+    parser.add_argument(
+        '-bs', '--block_size', required=True,
+        type=int, action='store',
+        help='Block size to be sampled with replacemnt for builidng bootstrapped'\
+        +' contigs.',
+    )
+    parser.add_argument(
+        '-r', '--replicates', required=True,
+        type=int, action='store',
+        help='Number of bootstrapped replicates to perform.'\
+    )
+    parser.add_argument(
+        '-p', '--path', required=False,
+        action='store', default='./',
+        help='Path for results and qc file (default = current working directory).',
+    )
+    parser.add_argument(
+        '-t', '--threads', required=False,
+        type=int, action='store', default=1,
+        help='Number of threads to use (default = 1).',
+    )
+
+    return parser.parse_args()
+
+
+def build_pop_dicc():
+    # Intialize an empty dictionary to store the focal population meta data.
+    pop_dicc = {
+        g_state['args'].p1_population: {'IND': [], 'IDX': []},
```

Comment:
Ok so I need a less @Jazpy, last time you told me that `c_` stood for current and now I use it all the time, but what does `g_` stand for?

Reply:
It's meant to mean global, since we store shared state by all thread in that dict. We should move away from global objects though, it's messy.